nsq/async.py

```diff
@@ -277,7 +277,7 @@ def _read_body(self, data):
             self.trigger(event.DATA, conn=self, data=data)
         except Exception:
             logger.exception('uncaught exception in data event')
-        self.io_loop.add_callback(self._start_read)
```

Comment:
@mreiferson thanks for digging a little further. I agree, and that addresses my previous concern.